tensorboard/plugins/graph/keras_util.py

```diff
@@ -41,18 +40,41 @@
 Please refer to https://github.com/tensorflow/tfjs-layers/blob/master/src/keras_format/model_serialization.ts
 for more complete definition.
 """
+import collections
+import six
+
 from tensorboard.compat.proto.graph_pb2 import GraphDef
 from tensorboard.compat.tensorflow_stub import dtypes
 
 
+# Please refer to below link for more information.
+# https://github.com/tensorflow/tensorflow/blob/1020739e1724346ef0088186f0253defaad3edb6/tensorflow/python/keras/engine/network.py#L1079-L1090
+InboundNodes = collections.namedtuple(
+    'InboundNodes', ['inbound_layer', 'node_index', 'tensor_id', 'kwargs'])
```

Comment:
What types of values are `node_index` and `tensor_id`? From the names,
I’d assume “int” and “locally unique string” respectively… but the
linked code in Keras actually calls them both `*_id`, whereas other code
in Keras calls them both `*_index`.

Reply:
I could see why that is confusing and I am confused in general too. Keras does not define the term firmly and like you pointed out earlier, some code refers to data as index and others refer to it as id.
I honestly dunno what to call them except I want to be consistent with _something_. Would you prefer index like the definition of `iterate_inbound`?
https://github.com/tensorflow/tensorflow/blob/1020739e1724346ef0088186f0253defaad3edb6/tensorflow/python/keras/engine/network.py#L1081

Reply:
I care most that we pick one and use it consistently. I tested this a
bit more and it _looks_ to me like both of these values are always
integers, and also TFJS calls them `NodeIndex` and `TensorIndex`
respectively, so maybe a slight preference for `index` over `id`?